java/src/main/native/src/ColumnViewJni.cu

```diff
@@ -51,4 +53,14 @@ new_column_with_boolean_column_as_validity(cudf::column_view const &exemplar,
   return deep_copy;
 }
 
+std::unique_ptr<cudf::column> generate_list_offsets(cudf::column_view const &list_length,
+                                                    rmm::cuda_stream_view stream) {
+  CUDF_EXPECTS(list_length.type().id() == cudf::type_id::INT32,
+               "Input column does not have type INT32.");
+
+  auto const begin_iter = list_length.template begin<cudf::size_type>();
+  auto const end_iter = list_length.template end<cudf::size_type>();
+
+  return cudf::strings::detail::make_offsets_child_column(begin_iter, end_iter, stream);
```

Comment:
There's no reason to use a detail API here. This is just a scan and can use `cudf::scan`.

Reply:
Hi @jrhemstad, we need to allocate a output buffer whose size is N + 1. Then, project the result of inclusive scan to the output buffer (with the offset 1).
Is `thrust::inclusive_scan` more flexible to do this job than `cudf::scan`?

Reply:
If I am not wrong, we need to run an extra memcpy with the `cudf::scan`.

Reply:
Humm, if we can't use the detail API then I hate to recommend this: just copy exactly the code from `cudf::strings::detail::make_offsets_child_column`.